spirv_common.hpp

```diff
@@ -296,6 +296,8 @@ struct SPIRType : IVariant
 		bool depth;
 		bool arrayed;
 		bool ms;
+		bool is_read = false;
```

Comment:
I think you can use the appropriate decorations for this. GLSL has the same deal with image load/store and readonly/writeonly. See DecorationNonWriteable and DecorationNonReadable. I'd like to avoid adding a new way to deal with this.